### PR TITLE
Added Template for NeoAlgoDoc

### LINKED ISSUE #60: Template for the format of the documentation
## 🚀 Feature

I had written a .txt file where it is already formatted in the required format from Introduction to credits.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation
To help the contributors with the right format and also saves time  

This can be added as Contributors Template.

@HarshCasper please assign this issue to me.

### PR DESCRIPTION
**Pull request by nagasrihith608**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
This a template for the documents. 

### Motivation
contributes can download this and use this so as they don't need to write/refer to the whole thing again just to save time and avoid mistakes in formatting.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes #60 
(Write your answer here.)

### CODE REVIEW

**Comment by nagasrihith608** [comment]
@HarshCasper please check the template once and suggest if any changes to be made. Thank you!

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you name it `ARTICLE-TEMPLATE.mdx` ? 

It would be visible to more contributors then.

**Comment by nagasrihith608** [comment]
@HarshCasper just renamed please check.

File: ARTICLE-TEMPLATE.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+```
+//write here
+```
+
+## Complexity Analysis
+
+```
+Time  complexity -
+Space complexity -
+```
```
Comment (HarshCasper) [inline ARTICLE-TEMPLATE.mdx:70]: Keep it as a bullet point rather than in triple backticks

File: ARTICLE-TEMPLATE.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
```
Comment (HarshCasper) [inline ARTICLE-TEMPLATE.mdx:1]: Add a comment on the top, asking users to copy the raw version of the file before applying them

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: ARTICLE-TEMPLATE.mdx (hunk 2 above)
Comment (nagasrihith608) [reply ARTICLE-TEMPLATE.mdx:1]: sorry did not get you can you elaborate

Comment (HarshCasper) [reply ARTICLE-TEMPLATE.mdx:1]: Basically we want the users to open the raw, un-rendered version of the file for their purpose. Not the rendered file. So having a comment at the top would be helpful

**Comment by nagasrihith608** [comment]
@HarshCasper I added the guideline please check now

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**